### ISSUE TITLE
Support linux appimage

### ISSUE THREAD

**Issue by jopemachine** [Izurii/easy-auto-launch#2]
- The target platform(s) the problem occurs on (Linux, Mac, Mac App Store, and or Windows): Linux
- Node version (run `node -v`): 12.22.1
- auto-launch version: 6.0.0
- The options you're passing to auto-launch: hidden=true
- Are you using NW.js or Electron? Electron
  - If so, which version?: Electron 12.0.1
  - If you're using Electron, are you using auto-launch from the main process or a renderer process? main-process
- Can you reproduce this with a simple Hello World example app? No
- If needs be, would you be able to provide us with a simple app (as simple as possible) which reproduces the problem? Yes

---

On Windows, and on Macos, easy-auto-launch works well.

But on Ubuntu using `Appimage`, easy-auto-launch launch below the screen (Electron) instead of my app.

![캡처](https://user-images.githubusercontent.com/18283033/120136932-3d608b00-c20e-11eb-9568-cdf2a3e4e5c5.PNG)

I think maybe the below issue, PR in the original repo could be related to this issue.

* https://github.com/Teamwork/node-auto-launch/pull/91
* https://github.com/Teamwork/node-auto-launch/issues/89

Thanks for remaking the awesome library!

**Comment by Izurii** [comment]
The problem I see in creating an auto launch for AppImage is that the path to the AppImage can be changed anytime. What I think we can do is something like this: The user opens the AppImage this generate the autorun entry but if later the user change the location of the AppImage the autorun entry will be invalid, but once the user reopen the app using the AppImage the autorun entry will be updated with the new location.

**Comment by jopemachine** [comment]
> The problem I see in creating an auto launch for AppImage is that the path to the AppImage can be changed anytime. What I think we can do is something like this: The user opens the AppImage this generate the autorun entry but if later the user change the location of the AppImage the autorun entry will be invalid, but once the user reopen the app using the AppImage the autorun entry will be updated with the new location.

Yes, because the path changes whenever the mount is done (reboot), the previously registered path is invalid.

How about passing `AppImage` file path itself to `Exec` instead of `process.execPath` in linux?

```
[Desktop Entry]
Type=Application
Exec=/home/jopebot/Applications/Arvis-0.2.0.AppImage
Hidden=false
...
```

This works as expected, but the user needs to pass the AppImage file path manually,

And if the file path is changed by the user, the entry file will be invalid.

(So I'm not sure this is the right way though)

**Comment by uparlange** [comment]
Hi,

I have the same message auto launching my app (https://github.com/uparlange/bing-wallpaper) with Windows 11.

It is working good with Windows 10 and MacOS (BigSur and Monterey).

![Capture d’écran 2021-11-07 090515](https://user-images.githubusercontent.com/5107041/140637347-2681a543-46f4-4d7f-b011-162cbabc399f.png)
.

**Comment by ghost** [comment]
Hello @Izurii,
Here is the fix: https://github.com/Netflix-Skunkworks/stethoscope-app/pull/126/commits/74ca61b422a9217caec2538fa8cb0e2f7b67f896

**Comment by ghost** [comment]
@Izurii, the condition should be added here: https://github.com/Izurii/easy-auto-launch/blob/cec445c2b80dbe4f7f5e1672004f65b6e27d341b/src/easy-auto-launch.ts#L122

Many thanks for this cleaner architecture!

**Comment by Izurii** [comment]
> Hello @Izurii, Here is the fix: [Netflix-Skunkworks/stethoscope-app@74ca61b](https://github.com/Netflix-Skunkworks/stethoscope-app/commit/74ca61b422a9217caec2538fa8cb0e2f7b67f896)

I'll be taking a look at that asap. Thank you for bringing a solution to this old bug :)
